docs/design/provisioning.md

```diff
@@ -0,0 +1,223 @@
+
+
+## Using ConfigMap
+
+### Admin namespace
+
+Provisioners will be configured using `ConfigMap`.  `ConfigMap` allows configuration to be stored in the API server, which allows dynamic updates, querying, watches, etc.  The alternative is file config, which requires the same machinery as the API server (versioning, watches, querying, etc), or a new 1st class API object.  `ConfigMap` perfectly suits the requirements of this proposal without requiring a new API type.
+
+`ConfigMap` objects live in a namespace, so the administrator will create a namespace specifically to hold provisioner configuration.  This "system namespace" may contain many types of ConfigMaps, not all of which are provisioners, so a specific label must be applied to provisioners to identify them.
+
+The  namespace and label are required by kube-controller-manager so that the provisioner controller can query the correct namespace and find the provisioner configs.
+
+Admins will pass the namespace and label to the controller via CLI flag:
+
+```
+CLI flags:
+
+--pv-provisioner-namespace=foobar
+--pv-provisioner-label="type=provisioner-config
+
+example usage:
+
+kubectl --namespace=system-namespace get configmap -l type=provisioner-config
+
+```
+
+
+### Provisioner Configuration
+
+Use case:  Configure provisioners for GCE Persistent Disks in two zones (east and west) and two volume types (standard and solid state).  The admin chooses to use the term "storage-class" as a stand-in for volume type (speed) and uses the labels "gold" and "silver" for fast and slow.
+
+A distinct set of labels on a provisioner is the "storage class".  The use case requires 4 provisioners: gold+east, gold+west, silver+east, silver+west.   Each provisioner's "parameters" map would contain the values for volume type and zone.  The plugin uses the parameters to create that one type of PersistentVolume.
+
+One configuration is shown below.  The remainder can be found [here](provisioning-examples.json).
+
+```
+  {
+    "kind": "ConfigMap",
+    "apiVersion": "v1",
+    "metadata": {
+      "name": "gold-east",
+      "namespace": "kubernetes.io/provisioners",
+      "labels": {
+        "type": "provisioner-config",
+        "storage-class": "gold",
+        "zone": "us-east"
+      }
+    },
+    "data": {
+      "plugin-name": "kubernetes.io/gce-pd",  
+      "parameters": "{'volumeType':'ssd', 'zone':'us-east'}"  // OPAQUE JSON MAP OF PARAMS AS DEFINED BY THE PLUGIN
+    }
+  }
+  
+```
+
+End users request specific storage by using a PersistentVolumeSelector on their claim.  This example shows a request for "gold" storage in the east zone.
+
+```
+{
+  "kind": "PersistentVolumeClaim",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "claim2",
+    "namespace": "end-user-namespace"
+  },
+  "spec": {
+    "accessModes": [
+      "ReadWriteOnce"
+    ],
+    "persistentVolumeSelector":{
+    	"matchLabels":{
+    		"storage-class":"gold",
+    		"zone":"us-east",
```

Comment:
This does not follow the example above. This should demonstrate the case where the cluster admin decided to bake the zone into his storage classes.

So `zone` should not be specified, and the `storage-class` should be `gold-east`.

The example below with selectors should show the counter example where the cluster admin did not bake the zone into the storage classes, and it is configurable via label.